### PR TITLE
wasm-tools: 1.216.0 -> 1.217.0

### PR DESCRIPTION
**Pull request by ereslibre**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ereslibre** [comment]
I tried to reproduce and I found a similar error on another test:

```
failures:

---- "tests/cli/wit-directory-output-in-deps-folder-with-multiple-version.wit" ----
failed test "tests/cli/wit-directory-output-in-deps-folder-with-multiple-version.wit"

Caused by:
    Broken pipe (os error 32)

---- "tests/cli/wit-directory-output-valid.wit" ----
failed test "tests/cli/wit-directory-output-valid.wit"

Caused by:
    Broken pipe (os error 32)


failures:
    "tests/cli/wit-directory-output-in-deps-folder-with-multiple-version.wit"
    "tests/cli/wit-directory-output-valid.wit"

test result: FAILED. 104 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.12s

error: test failed, to rerun pass `-p wasm-tools --test cli`
error: builder for '/nix/store/8giijrm17iphm9hyrwf6vxm0n6cg5rdq-wasm-tools-1.217.0.drv' failed with exit code 101;
       last 10 log lines:
       >     Broken pipe (os error 32)
       >
       >
       > failures:
       >     "tests/cli/wit-directory-output-in-deps-folder-with-multiple-version.wit"
       >     "tests/cli/wit-directory-output-valid.wit"
       >
       > test result: FAILED. 104 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.12s
       >
       > error: test failed, to rerun pass `-p wasm-tools --test cli`
       For full logs, run 'nix log /nix/store/8giijrm17iphm9hyrwf6vxm0n6cg5rdq-wasm-tools-1.217.0.drv'.
```

If I find the time, I'll have a look at what's going on.

**Comment by ereslibre** [comment]
I have submitted https://github.com/bytecodealliance/wasm-tools/issues/1820 to keep track of this issue. I'm going to update the flags so that we force `--test-threads=1` for now.

**Comment by ereslibre** [comment]
Updated the PR to set test threads to 1 for aarch64. I was not able to reproduce the issue with this setting.

File: pkgs/tools/misc/wasm-tools/default.nix [hunk 1]
```diff
@@ -1,25 +1,28 @@
 { lib
 , rustPlatform
 , fetchFromGitHub
+, stdenv
```
Comment (NickCao) [inline pkgs/tools/misc/wasm-tools/default.nix:4]: unused parameter, otherwise LGTM.

Comment (ereslibre) [reply pkgs/tools/misc/wasm-tools/default.nix]: Oops,leftover. Fixed. Thanks!

**Review by NickCao — APPROVED** [review]

**Pull request merged**